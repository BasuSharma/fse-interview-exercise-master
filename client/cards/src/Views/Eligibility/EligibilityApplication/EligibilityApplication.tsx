import React from "react";
import { useFormik } from "formik";
import styled from "styled-components";
import View from "../../../DesignSystem/View";
import FormInput from "../../../DesignSystem/Form/FormInput";
import SubmitButton from "../../../DesignSystem/Form/SubmitButton";
import Title from "../../../DesignSystem/Title";
import * as Yup from 'yup';
import {INPUT_NAME_ERROR_MSG,INPUT_EMAIL_ERROR_MSG,INPUT_ADDRESS_ERROR_MSG} from "../../../Configuration/ApplicationConstants"

const FormWrapper = styled.div`
  flex: 1 1 auto;
  width: 100%;
`;

interface FormValues {
  name: string;
  email: string;
  address: string;
}

// Component to collect user data in Formik forms and send details to parent conatiner
// Uses submit action from props provided funciton
const EligibilityApplication = (props: { checkApplicantEligiblity: any }) => {
  const formik = useFormik<FormValues>({
    initialValues: {
      name: "",
      email: "",
      address: "",
    },
    validationSchema: Yup.object({
      name: Yup.string()
        .min(3,INPUT_NAME_ERROR_MSG)
        .max(35, INPUT_NAME_ERROR_MSG)
        .required(INPUT_NAME_ERROR_MSG),
      email: Yup.string().email(INPUT_EMAIL_ERROR_MSG).required(INPUT_EMAIL_ERROR_MSG),
      address: Yup.string()
        .min(5,INPUT_ADDRESS_ERROR_MSG)
        .max(55, INPUT_ADDRESS_ERROR_MSG)
        .required(INPUT_ADDRESS_ERROR_MSG),
    }),
    onSubmit: (values) => props.checkApplicantEligiblity(values)
    ,
  });

  return (
    <View>
      <FormWrapper>
        <Title>Cards</Title>
        <form onSubmit={formik.handleSubmit}>
          <FormInput
            type="text"
            id="name"
            {...formik.getFieldProps('name')}
            placeholder="Name"
          />
          {formik.touched.name && formik.errors.name ? (
         <div>{formik.errors.name}</div>
          ) : null}
          <FormInput
            type="email"
            id="email"
            {...formik.getFieldProps('email')}
            placeholder="Email"
          />
          {formik.touched.email && formik.errors.email ? (
            <div>{formik.errors.email}</div>
            ) : null}

          <FormInput
            type="text"
            id="address"
            {...formik.getFieldProps('address')}
            placeholder="Address"
          />
          {formik.touched.address && formik.errors.address ? (
         <div>{formik.errors.address}</div>
          ) : null}
          <SubmitButton text="Submit" />
        </form>
      </FormWrapper>
    </View>
  );
};

export default EligibilityApplication;


