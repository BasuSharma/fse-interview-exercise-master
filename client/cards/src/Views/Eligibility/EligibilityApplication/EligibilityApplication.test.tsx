import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import EligibilityApplication from './EligibilityApplication';

Enzyme.configure({ adapter: new Adapter() })


let mockFn = jest.fn();

 
it(" EligibilityApplication Component renders correctly", () => {
  const testApplicaitonFormcomponent = shallow(<EligibilityApplication checkApplicantEligiblity={mockFn} />);
  expect(toJson(testApplicaitonFormcomponent)).toMatchSnapshot();
});
