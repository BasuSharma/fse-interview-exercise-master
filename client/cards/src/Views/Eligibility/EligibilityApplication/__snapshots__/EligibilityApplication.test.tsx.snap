// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` EligibilityApplication Component renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.h1>
      Cards
    </styled.h1>
    <form
      onSubmit={[Function]}
    >
      <FormInput
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
      <FormInput
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
      <FormInput
        id="address"
        name="address"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Address"
        type="text"
        value=""
      />
      <SubmitButton
        text="Submit"
      />
    </form>
  </styled.div>
</styled.div>
`;
